docs/06-actors-actions.md

```diff
@@ -0,0 +1,453 @@
+
+```ts
+/**
+ * DANGER: This is for advanced users to totally override draw logic.
+ */
+public draw(ctx: CanvasRenderingContext2D, delta: number) {
+
+  // Invoke the core logic
+  // Or leave it out to completely override
+  super.draw(ctx, delta);
+
+  // custom drawing
+  ctx.lineTo(...);
+}
+```
+
+### Post-draw
+
+[[Actor.onPostDraw]] is run _after_ [[Actor.draw]] and will draw in the current frame.
+
+<docs-note>This is the recommended method to override since Excalibur has run its core draw logic and you can now customize what gets drawn during the current frame.</docs-note>
+
+```ts
+/**
+ * RECOMMENDED: This is run at the end of the draw loop. Usually
+ * this is what you want!
+ */
+public onPostDraw(ctx: CanvasRenderingContext2D, delta: number) {
+
+  /* perform custom draw */
+  ctx.lineTo(...);
+}
+```
+
+### Debug draw
+
+[[Actor.debugDraw]] provides a way to customize the way Excalibur shows actors when [debugging is enabled](/docs/intro#enabling-debug-mode). Override it to add or replace any draw logic with your own:
+
+```ts
+/**
+ * This is run at the end of the draw loop
+ */
+public debugDraw(ctx: CanvasRenderingContext2D) {
+
+   // Call core logic
+   super.debugDraw(ctx, delta);
+
+   // Add custom debug drawing logic
+   ctx.lineTo(...);
+}
+```
+
+### Adding actors to the scene
+
+For an [[Actor]] to be drawn and updated, it needs to be part of the "scene graph".
+The [[Engine]] provides several easy ways to quickly add/remove actors from the
+current scene.
+
+```js
+var game   = new ex.Engine(...);
+var player = new ex.Actor();
+var enemy  = new ex.Actor();
+// add them to the "root" scene
+game.add(player);
+game.add(enemy);
+// start game
+game.start();
+```
+
+You can also add actors to a [[Scene]] instance specifically.
+
+```js
+var game = new ex.Engine()
+var level1 = new ex.Scene()
+var player = new ex.Actor()
+var enemy = new ex.Actor()
+// add actors to level1
+level1.add(player)
+level1.add(enemy)
+// add level1 to the game
+game.add('level1', level1)
+// start the game
+game.start()
+// after player clicks start game, for example
+game.goToScene('level1')
+```
+
+## Collision Detection
+
+By default Actors do not participate in collisions. If you wish to make
+an actor participate, you need to switch from the default [[CollisionType.PreventCollision|prevent collision]]
+to [[CollisionType.Active|active]], [[CollisionType.Fixed|fixed]], or [[CollisionType.Passive|passive]] collision type.
+
+<docs-note>For more information on collisions, see [Physics](/docs/physics).</docs-note>
+
+```ts
+public Player extends ex.Actor {
+   constructor() {
+      super();
+      // set preferred CollisionType
+      this.body.collider.type = ex.CollisionType.Active;
+   }
+}
+
+// or set the collisionType
+
+const actor = new ex.Actor();
+actor.body.collider.type = ex.CollisionType.Active;
+```
+
+## Traits
```

Comment:
Traits seem super useful, an example of a custom trait would be highly helpful.

Reply:
@eonarheim Traits are going away anyway and being replaced with Components, right? We may as well wait for the new refactor to expand on that.

Reply:
That's the plan, `Traits` and `Components` are very similar ideas. Our end goal is to make things as composable as possible :+1:

I agree, nothing has hit master yet, and `Traits` will be around for a little while longer before we have `Components` of equivalent functionality.